src/Datagrid/Datagrid.php

```diff
@@ -176,8 +179,20 @@ static function ($value) {
         $this->bound = true;
     }
 
+    /**
+     * NEXT_MAJOR: Use `ChainableFilterInterface` as type declaration for argument 1.
+     */
     public function addFilter(FilterInterface $filter)
     {
+        if (!$filter instanceof ChainableFilterInterface) {
```

Comment:
Why is it now required ?

This interface is highly related to the search. IMHO it shouldn't be a required for this method.
The datagrid should still works with simple FilterInterface.

In a similar way, I introduced the SearchableFilterInterface
https://github.com/sonata-project/SonataAdminBundle/pull/7275/files#diff-78b56f87070cbd240d72b09e6194932274323e06a06e985c148dab0b08a6d881R94

After the merge of my PR, I would say that if a filter does not implement the SearchableFilterInterface, it doesn't have to be a ChainableFilterInterface since it won't be used in the search.

Maybe the use of the interface could be improve after my PR.
Maybe only one interface is needed and you can add the method to the SearchableFilterInterface ; or the SearchableFilterInterface should extends your interface ?